apps/cli/src/runAstgrepCodemod.ts

```diff
@@ -0,0 +1,62 @@
+import fs from 'fs';
+import { execSync } from 'node:child_process';
+import { readFile } from 'node:fs/promises';
+import path from 'path';
+import * as yaml from 'js-yaml';
+import { PrinterBlueprint } from './printer.js';
+
+export const runAstgrep = async (
+	printer: PrinterBlueprint,
+	rulePath: string,
+	targetDirectory: string,
+): Promise<void> => {
+	const yamlString = await readFile(rulePath, { encoding: 'utf8' });
+	const yamlObject = yaml.load(yamlString);
+	const language = languageToExtension(yamlObject.language);
+	printer.printConsoleMessage(
+		'info',
+		`Executing ast-grep for language : ${language}`,
+	);
+
+	// Function to recursively iterate over files in the directory
+	const iterateFiles = async (dirPath: string) => {
+		const entries = await fs.promises.readdir(dirPath, {
+			withFileTypes: true,
+		});
+
+		// Iterate over each entry in the directory
+		for (const entry of entries) {
+			const entryPath = path.join(dirPath, entry.name);
+			// Check if the entry is a directory
+			if (entry.isDirectory()) {
+				// Recursively call the function for subdirectories
+				await iterateFiles(entryPath);
+			} else if (entry.isFile()) {
+				// If the entry is a file, log its path
+				const fileExtension = path.extname(entryPath).slice(1);
+				if (fileExtension !== language) {
+					continue;
+				}
+				const astCommand = `sg scan -r ${rulePath} ${entryPath} -U`;
```

Comment:
will this work on `Windows`? i think we need to add quotes to the args to make it work in Windows cmd. I will check later.

Reply:
Updated the code